### DISCUSSION TITLE
Remove ( var ) declarations

### DISCUSSION THREAD

**Discussion by Eprahim-taha** [modular/modular#1506] [Q&A]
You could remove the keyword  ( var ) Why don't we simplify things?

var a: String = "Hello Mojo 🔥"     ❌

a: String = "Hello Mojo 🔥"    ✅.       # Here Mojo Compiler 🔥 will recognize the type of the 
                                                                   variable at compile time  

If you write the variable type as we explained in the previous example, the variable type will be recognized at compile time. If you do not write the variable type, the variable type will be recognized at run time For example 

a =  "Hello Mojo 🔥"   ✅       # here Mojo Compiler 🔥 will recognize the type of the variable 
                                                     at runtime

**Comment by Eprahim-taha** [comment]
This is just a suggestion, not based on complete knowledge of the Mojo programming language 🔥 I'm just a novice programmer. If anyone has any comments on this suggestion, you can discuss it here.

**Reply by Sharktheone** [reply]
Have you seen #1456? There they are basically talking about that topic. So, you should probably repost your message there. 

From my perspective I generally don't quite like it to have almost the same syntax for declaring a variable and overwriting / updating it.

**Comment by lattner** [comment]
Right, let's discuss this in 1456
